sim/hunter/items.go

```diff
@@ -355,10 +358,146 @@ func init() {
+				character.MoveSpeed *= 1.40
+			},
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				character.MoveSpeed /= 1.40
+			},
+		})
+
+		core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
+			Name:              "Kestrel Trigger",
+			Callback:          core.CallbackOnSpellHitDealt,
+			Outcome:           core.OutcomeLanded,
+			ProcMask:          core.ProcMaskMeleeMH,
+			SpellFlagsExclude: core.SpellFlagSuppressWeaponProcs,
+			PPM:               1.0,
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				kestrelAura.Activate(sim)
+			},
+		})
+	})
+
+	// https://www.wowhead.com/classic/item=231288/renatakis-charm-of-ravaging
+	core.NewItemEffect(RenatakisCharmOfRavaging, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		lockedIn := character.RegisterAura(core.Aura{
+			Label: "Locked In",
+			ActionID: core.ActionID{SpellID: 468388},
+			Duration: time.Second * 20,
+			MaxStacks: 3,
+			OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+				if spell.SpellCode == SpellCode_HunterCarve || spell.SpellCode == SpellCode_HunterMultiShot {
+					spell.CD.Reset()
+					aura.RemoveStack(sim)
+				}
+			},
+		})
+
+		spell := character.GetOrRegisterSpell(core.SpellConfig{
+			ActionID: core.ActionID{SpellID: 468388},
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    character.NewTimer(),
+					Duration: time.Minute * 2,
+				},
+			},
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				lockedIn.Activate(sim)
+				lockedIn.SetStacks(sim, 3)
+			},
+		})
+
+		character.AddMajorCooldown(core.MajorCooldown{
+			Spell: spell,
+			Type:  core.CooldownTypeDPS,
+		})
+	})
+
+	// https://www.wowhead.com/classic/item=230237/arcane-infused-gem
+	core.NewItemEffect(ArcaneInfusedGem, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		arcaneDetonation := character.RegisterSpell(core.SpellConfig{
+			ActionID: 		core.ActionID{SpellID: 467447},
+			SpellSchool: 	core.SpellSchoolArcane,
+			DefenseType: 	core.DefenseTypeMagic,
+			ProcMask: 		core.ProcMaskSpellDamage,
+			Flags: 			core.SpellFlagNoOnCastComplete | core.SpellFlagPassiveSpell,
+
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+			BonusCoefficient: 0,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				for _, aoeTarget := range sim.Encounter.TargetUnits {
+					damage := sim.Roll(185, 210)
+					spell.CalcAndDealDamage(sim, aoeTarget, damage, spell.OutcomeMagicCrit)
+				}
+			},
+		})
+
+		arcaneInfused := character.RegisterAura(core.Aura{
+			Label: "Arcane Infused",
+			ActionID: core.ActionID{SpellID: 467446},
+			Duration: time.Second * 15,
+			OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+				maxTargetsPerCast := int32(5)
```

Comment:
Can you assign this outside of the aura so that it doesn't have to be on every cast?

Reply:
Actually it could be done like this too to not have to check the min every time below:
```
maxTargetsPerCast := min(sim.Environment.GetNumTargets, 5)
```

Reply:
ended up splitting out the max targets for multishot and carve and setting them on initialization rather than on cast complete